diff --git a/src/restore.c b/src/restore.c
index a78783d..89b5b11 100644
--- a/src/restore.c
+++ b/src/restore.c
@@ -3767,7 +3767,7 @@ static int restore_send_firmware_updater_data(struct idevicerestore_client_t* cl
 	}
 
 	plist_get_string_val(p_updater_name, &s_updater_name);
-	logger_dump_plist(LL_DEBUG, p_info, 1);
+	logger_dump_plist(LL_DEBUG, arguments, 1);
 
 	if (strcmp(s_updater_name, "SE") == 0) {
 		fwdict = restore_get_se_firmware_data(client, p_info, arguments);
@@ -3780,9 +3780,12 @@ static int restore_send_firmware_updater_data(struct idevicerestore_client_t* cl
 		plist_t p_info_yonkers = plist_dict_get_item(p_info, "YonkersDeviceInfo");
 		plist_t p_info_jasmine = plist_dict_get_item(p_info, "JasmineIR1DeviceInfo");
                 plist_t p_info_yonkers_ir1 = plist_dict_get_item(p_info, "YonkersIR1,DeviceInfo");
-		if (PLIST_IS_DICT(p_info_yonkers) || PLIST_IS_DICT(p_info_yonkers_ir1)) {
+		if (PLIST_IS_DICT(p_info_yonkers)) {
 			fwtype = "Yonkers";
 			fwdict = restore_get_yonkers_firmware_data(client, p_info_yonkers, arguments);
+		} else if (PLIST_IS_DICT(p_info_yonkers_ir1)) {
+			fwtype = "YonkersIR1";
+			fwdict = restore_get_generic_firmware_data(client, p_info_yonkers_ir1, arguments);
 		} else if (PLIST_IS_DICT(p_info_jasmine)) {
 			fwtype = "Jasmine";
 			fwdict = restore_get_generic_firmware_data(client, p_info_jasmine, arguments);
